### PR TITLE
Hdrp/deferred shadow quality multi compile

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
Same as https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/6328

### CODE REVIEW

**Comment by FrancescoC-unity** [comment]
Reopening this as migration did not happen

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/HDShadowAlgorithms.hlsl [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #define FIXED_UNIFORM_BIAS (1.0f / 65536.0f)
 
 // We can't use multi_compile for compute shaders so we force the shadow algorithm
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/HDShadowAlgorithms.hlsl:12]: remove this comment :)

**Comment by sebastienlagarde** [comment]
@FrancescoC-unity . Various thing:
- can you merge master
- can you add a section in upgrade guide saying what will happen (as people could have setup shadow filtering to low or high and it wasn't affecting their rendering)
- can you update the documentation on the shadow pass that talk about using config to tweak shadow quality
- add a TODO in the shader stripper to say we must strip shadow compute (so we don't forget)

For QA:
- Please test shadow mask, be sure they works (should be cover mostly by yamato)
- Build player and check that shadow filtering selected is the right one in player (test with different one)
- check that when you enable debug display mode thing still work
- check that when you disable material or light classification in DebugLoop of Framesettings, thing still work

**Comment by sebastienlagarde** [comment]
Note: build player time will increase but it should still faiirly manageable compare to lit.shader, and Juho is working on the stripping part.

File: com.unity.render-pipelines.high-definition/Documentation~/Upgrading-from-2020.1-to-2020.2.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Upgrading HDRP from Unity 2020.1 to Unity 2020.2
+
+In the High Definition Render Pipeline (HDRP), some features work differently between major versions of Unity. This document helps you upgrade HDRP from Unity 2020.1 to 2020.2.
+
+## Shadows
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Documentation~/Upgrading-from-2020.1-to-2020.2.md:5]: Note, I named this generically Shadows as when the shadow caching rewrite lands I might need to add something here.

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/Deferred.compute [hunk 3]
```diff
@@ -34,6 +34,8 @@
 
 #pragma multi_compile _ SHADOWS_SHADOWMASK
 
+#pragma multi_compile SHADOW_LOW SHADOW_MEDIUM SHADOW_HIGH
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/Deferred.compute:37]: you need to do this also for deferred.shader and tiledeferred.shader

**Comment by sebastienlagarde** [comment]
For QA: Please test also with deferredTile and deferred if thing work with shadow.
Please check also if the selected filtering quality work correctly in the player. thanks

**Comment by sebastienlagarde** [comment]
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/HDRP%252Fdeferred-shadow-quality-multi-compile/.yamato%252Fupm-ci-hdrp.yml%2523All_HDRP_fast-2020.1

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/Deferred.compute (hunk 3 above)
Comment (FrancescoC-unity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/Deferred.compute:37]: Ha! Yes I always forget about those two.  Will do!

**Pull request merged**